client/components/jetpack/connection-health/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import Notice from 'calypso/components/notice';
+import NoticeAction from 'calypso/components/notice/notice-action';
+import { JETPACK_CONNECTION_HEALTHY } from 'calypso/state/action-types';
+import { useCheckJetpackConnectionHealth } from './use-check-jetpack-connection-health';
+
+interface Props {
+	siteId: number;
+}
+
+export const JetpackConnectionHealthBanner = ( { siteId }: Props ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const [ isErrorCheckJetpackConnectionHealth, setIsCheckJetpackConnectionHealth ] =
+		useState( false );
+
+	const { data: jetpackConnectionHealth, isLoading: isLoadingJetpackConnectionHealth } =
+		useCheckJetpackConnectionHealth( siteId, {
+			onError: () => {
+				setIsCheckJetpackConnectionHealth( true );
+			},
+			onSuccess: () => {
+				dispatch( {
+					type: JETPACK_CONNECTION_HEALTHY,
+					siteId,
+				} );
+			},
+		} );
+
+	if (
+		isLoadingJetpackConnectionHealth ||
+		isErrorCheckJetpackConnectionHealth ||
+		jetpackConnectionHealth?.is_healthy
+	) {
+		return;
+	}
+
```

Comment:
At this point, can we determine if the jetpack connection is failing just by checking the API response?

Reply:
If `is_healthy` is `true` it means API is ok and we don't want to display a notice.